src/Compilers/Core/Portable/Compilation/TypeInfo.cs

```diff
@@ -27,23 +27,23 @@ namespace Microsoft.CodeAnalysis
         public ITypeSymbol ConvertedType { get; }
 
         // PROTOTYPE(nullable-api): Doc Comment
-        public Nullability ConvertedNullability { get; }
+        public NullabilityInfo ConvertedNullabilityInfo { get; }
 
-        internal TypeInfo(ITypeSymbol type, ITypeSymbol convertedType, Nullability nullability, Nullability convertedNullability)
+        internal TypeInfo(ITypeSymbol type, ITypeSymbol convertedType, NullabilityInfo nullabilityInfo, NullabilityInfo convertedNullabilityInfo)
             : this()
         {
             this.Type = type;
-            this.Nullability = nullability;
+            this.NullabilityInfo = nullabilityInfo;
             this.ConvertedType = convertedType;
-            this.ConvertedNullability = convertedNullability;
+            this.ConvertedNullabilityInfo = convertedNullabilityInfo;
         }
 
         public bool Equals(TypeInfo other)
         {
             return object.Equals(this.Type, other.Type)
                 && object.Equals(this.ConvertedType, other.ConvertedType)
-                && object.Equals(this.Nullability, other.Nullability)
-                && object.Equals(this.ConvertedNullability, other.ConvertedNullability);
+                && object.Equals(this.NullabilityInfo, other.NullabilityInfo)
+                && object.Equals(this.ConvertedNullabilityInfo, other.ConvertedNullabilityInfo);
```

Comment:
`NullabilityInfo` should implement `IEquatable<NullabilityInfo>` and the implementation should be used here to avoid boxing.